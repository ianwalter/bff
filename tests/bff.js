const { test, run } = require('..')

const config = { timeout: 5000, plugins: ['tests/helpers/plugin.js'] }
const toName = ({ name, err }) => name + (err ? `: ${err}` : '')

test('bff', async ({ expect }) => {
  const result = await run(config)
  expect(result.filesRegistered).toBe(5)
  expect(result.testsRegistered).toBe(26)
  expect(result.testsRun).toBe(26)
  expect(result.passed.length).toBe(15)
  expect(result.passed.map(toName).sort()).toMatchSnapshot()
  expect(result.failed.length).toBe(8)
  expect(result.failed.map(toName).sort()).toMatchSnapshot()
  expect(result.skipped.length).toBe(2)
  expect(result.skipped.map(toName).sort()).toMatchSnapshot()
})

test('bff --failFast', async ({ expect }) => {
  const result = await run({ ...config, failFast: true })
  expect(result.testsRegistered).toBeGreaterThan(0)
  expect(result.testsRun).toBeGreaterThan(0)
  expect(result.failed.length).toBe(1)
})
