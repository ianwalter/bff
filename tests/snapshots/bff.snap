// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bff 1`] = `
Array [
  "A test with a multiline name",
  "arithmetic",
  "beforeEach",
  "dom snapshot",
  "manual pass",
  "manual pass",
  "only",
  "only with test specified after name",
  "parseInt after a 1 second timeout",
  "registration [ONE]",
  "registration [TWO]",
  "second snapshot pass",
  "snapshot pass",
  "strict equality",
  "tags second call",
]
`;

exports[`bff 2`] = `
Array [
  "afterEach: manual failure",
  "manual fail with reason: because reasons",
  "manual fail: manual failure",
  "no assertions: no assertions made",
  "parseInt with the wrong base: [2mexpect([22m[31mreceived[39m[2m).toBe([22m[32mexpected[39m[2m) // Object.is equality[22m

Expected: [32m2[39m
Received: [31mNaN[39m",
  "snapshot fail: [2mexpect([22m[31mreceived[39m[2m).[22mtoMatchSnapshot[2m()[22m

Snapshot name: \`snapshot fail 1\`

[32m- Snapshot[39m
[31m+ Received[39m

[33m@@ -1,8 +1,8 @@[39m
[2m  \\"<html>[22m
[2m    <head>[22m
[32m-     <title>Demo</title>[39m
[31m+     <title>Dema</title>[39m
[2m    </head>[22m
[2m    <body>[22m
[2m      <main>[22m
[2m        <h1>Demo</h1>[22m
[2m      </main>[22m",
  "tags: [2mexpect([22m[31mreceived[39m[2m).[22mtoContain[2m([22m[32mexpected[39m[2m) // indexOf[22m

Expected value: [32m3[39m
Received array: [31m[1, 2][39m",
  "timeout: Promise timed out after 5000 milliseconds",
]
`;

exports[`bff 3`] = `
Array [
  "skip no assertions",
  "skip with test specified after name",
]
`;
